### ISSUE TITLE
Erroneously defined trunk vlans on interface trigger internal error

### ISSUE THREAD

**Issue by jbemmel** [ipspace/netlab#575]
When I try a [topology](https://github.com/jbemmel/netlab/blob/sros-evpn-mpls-update/tests/integration/evpn/mpls-vlan-bundle-trunks.yml) containing this:
```
vrfs:
  bundle_red_blue: # 2 vlans
    evpn:
  bundle_green: # single vlan
    evpn:

vlans:
  red:
    mode: bridge
    vrf: bundle_red_blue
  blue:
    mode: bridge
    vrf: bundle_red_blue
  green:
    mode: bridge
    vrf: bundle_green

...

links:
- h1:
  s1:
    vlan.trunk: [ red, blue, green ]
```

I get an error:
```
Fatal error in netlab: Internal error in map_trunk_vlans: VLAN blue should be already defined on node h1
```

'create_svi_interfaces' only operates on access vlans - so where in the code should trunk vlans get processed/copied in?

**Comment by ipspace** [comment]
I agree the error shouldn't happen (after all, it does say "internal error" which means "we messed up pretty badly") but I'd suspect the root cause is indentation ;)

What your topology is saying is "on S1 I have vlan trunk with red/blue/green, on H1 I have non-tagged traffic". I agree that should have been caught early in the VLAN validation code, will fix.

**Comment by jbemmel** [comment]
Yes, you're correct - 2 spaces too many

Error also happens on released version, not specific to dev-1.4

**Comment by ipspace** [comment]
BTW, the minimum topology needed to reproduce this behavior is:

```
defaults.device: eos

vlans:
  red:
  blue:

module: [ vlan ]
nodes: [ h1, s1 ]


links:
- h1:
  s1:
    vlan.trunk: [ red, blue ]
```

It would be really nice if you could post minimum topology when opening issues ;)

**Comment by ipspace** [comment]
This was a nasty one. First I implemented a bunch of tests to check whether we need native VLAN on a trunk link (your topology did, so it threw an error) and then whether every VLAN in a trunk is used by at least two nodes. At least the VLAN code is more robust now ;)

However, `rt-vlan-trunk-partial-overlap.yml` would still trigger a crash. It turns out that the VLANs on a physical trunk get accumulated into link **vlan.trunk** attribute (we need that to tick off member links) which was then merged with interface data, so all nodes should have all VLANs in the trunk -- clearly now what we asked **netlab** to do -- but fortunately we had the final sanity check that caused an abort.

In the end, I had to implement yet another tweak in `links.py`: do not copy some module attributes (`[ vlan ]`) from links to interfaces as they have been copied before by the module itself.
